source/common/router/config_impl.cc

```diff
@@ -1117,6 +1109,17 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_port = redirect_config_->port_redirect_.c_str();
   } else {
     final_port = "";
+    const absl::string_view current_path = headers.getHostValue();
```

Comment:
Out of curiosity, this is a direct response but not necessarily a redirect when you get here I think, what's to stop this applying even if it's not a redirect?